docs/DeveloperGuide.adoc

```diff
@@ -257,11 +292,104 @@ We are using `java.util.logging` package for logging. The `LogsCenter` class is
+* **Alternative 2:** Modify the Amount in the grocery item.
+** Pros: Will be more efficient and use less memory.
+** Cons: We must ensure that the implementation is correct.
+
+
 [[Implementation-Configuration]]
 === Configuration
 
 Certain properties of the application can be controlled (e.g user prefs file location, logging level) through the configuration file (default: `config.json`).
 
+=== GroceryList Feature
+
+=== WasteList Feature
+
+=== Edit Template Item feature
+The edit template item mechanism is facilitated by `UniqueTemplateItems`, `TemplateList` and `TemplateToBeShown`.
+
+The `TemplateList` is an observable list of `UniqueTemplateItems` while the UniqueTemplateItems contains an observable list
+of template items. The `TemplateToBeShown` is a instance of the `UniqueTemplateItems` that is used to show the current template
+that is visible on the user interface.
+
+To edit the individual templates stored in the `TemplateList`, the command retrieves the template as well as the item to be edited.
+Retrieving the existing item allows for retrieval of unchanged data field. The template item at the specified index will then be
+overwritten with a new edited TemplateItem.
+
+Similarly, the `UniqueTemplateItems` object at the specified index will also be overwritten by a new `UniqueTemplateItems`
+containing the edited template item with unchanged template items already transferred into the new `UniqueTemplateItems` object.
+The model is updated with the new `TemplateList`, as well as the new `TemplateToBeShown`.
+
+The creation of new objects to override the previous is necessary as template items and templates are stored as immutable objects in the `UniqueTemplateItems`
+and `TemplateList` respectively, and therefore cannot be edited directly.
+
+* `Model#getFilteredTemplateList()` -- Gets an unmodifiable template list.
+* `Model#setTemplate()` -- Replaces the previous template object with the new template object.
+* `Model#setShownTemplate()` -- Replaces the previous templateToBeShown with the new template so that the edited template is visible in the UI.
+
+These operations are exposed in the `Model` interface as `Model#getFilteredTemplateList()`, `Model#setTemplate()`
+`Model#setShownTemplate()` and `Model#updateFilteredTemplateToBeShown()` respectively.
+
+NOTE: When the user launches the application for the first time. The `TemplateToBeShown` will be initialized however it will not
+be visible until a templateItemCommand is executed. This can be done using an AddTemplateItemCommand, a DeleteTemplateItemCommand,
+a ListTemplateItemCommand, or an EditTemplateItemCommand.
+
+The following activity diagram summarizes what happens when a user executes a new command related to managing of template items:
+
+image::TemplateItemCommandUIActivityDiagram.png[]
+
+The following sequence diagram shows how the edit template item operation works for the logic component:
+
+image::EditTemplateItemSequenceDiagram2.png[]
```

Comment:
Should use `result:CommandResult` instead because `result` is passed back to `LogicManager`.